### ISSUE TITLE
Is it possible to change environment variable for web.config during Web deploy?

### ISSUE THREAD

**Issue by neman** [aspnet/Docs#10513]
I'm publishing app using Web Deploy (VS Project right click -&gt; publish -&gt; web deploy publish profile)
Can you please add instructions how to Set environment variables for web.config during web deploy?
 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe69b3a9-896f-e189-0c2c-f87ff735df9c
* Version Independent ID: 97c8549b-66ad-0a23-1971-784fb923042a
* Content: [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2#setting-environment-variables)
* Content Source: [aspnetcore/host-and-deploy/aspnet-core-module.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/host-and-deploy/aspnet-core-module.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @neman ... Thanks for asking. When the engineers were discussing it with the community on the engineering repo, they didn't open a docs issue at that time. So, let's ask now.

btw --- Just to recap for the *environment ONLY*: In the Web Deploy profile (pubxml), provide the following ...

```xml
<EnvironmentName>YourCustomEnvironment</EnvironmentName>
```

... or to use an MSBuild property, use `$(EnvironmentName)` within the project file.

**WRT env vars broadly via transforms ...** @vijayrkn ...

Is [vijayrkn/webconfigtransform](https://github.com/vijayrkn/webconfigtransform/blob/master/README.md) officially supported for 2.2?

* If so, then I'll get this scheduled.
* If it isn't supported yet, @neman can use it unsupported, and we'll wait on documenting until you pull the trigger.

There's also a fair amount of ❤️ out there for [nil4/dotnet-transform-xdt](https://github.com/nil4/dotnet-transform-xdt), and the features might not exactly match `vijayrkn/webconfigtransform` ... probably worth taking a look at if `vijayrkn/webconfigtransform` doesn't fit your scenario.

Going back to `<EnvironmentName>` and `$(EnvironmentName)`, let's surface in the following locations where appropriate:

* [Fundamentals: Environments (Set the environment: Windows section)](https://docs.microsoft.com/aspnet/core/fundamentals/environments#windows)
* [VS Publish Profiles (Publish to an MSDeploy endpoint section)](https://docs.microsoft.com/aspnet/core/host-and-deploy/visual-studio-publish-profiles#publish-to-an-msdeploy-endpoint-from-the-command-line)
* [Host & Deploy: ANCM (Set env vars section)](https://docs.microsoft.com/aspnet/core/host-and-deploy/aspnet-core-module#setting-environment-variables)

I'll work that separately (on a new issue that I'll open) from the broader case that you're asking about if we end up using this issue to document `vijayrkn/webconfigtransform`.

**Comment by vijayrkn** [comment]
> Is vijayrkn/webconfigtransform officially supported for 2.2?

Yes, the web.config transform support is officially supported for 2.2. Also this should have all the support similar to the global tool.

**Comment by guardrex** [comment]
Thanks @vijayrkn ... I'll work it on the issue and see about getting it done on the next sprint.

I'll open `<EnvironmentName>` and `$(EnvironmentName)` as a separate issue, and I'll try to get that on this sprint.
